src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,96 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_RESIZE = 2;
+    private Node<K, V>[] table;
+    private int currentLoad;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        currentLoad = (int) (DEFAULT_LOAD_FACTOR * DEFAULT_INITIAL_CAPACITY);
+    }
+
+    private int hash(K key) {
+        int h;
+        return (key == null) ? 0 : (h = key.hashCode()) ^ (h >>> 16);
+    }
+
+    private int getIndex(K key) {
+        return hash(key) & (table.length - 1);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int index = getIndex(key);
+        Node<K, V> newElement = new Node<>(hash(key), key, value, null);
+        Node<K, V> element = table[index];
+        if (element == null) {
+            table[index] = newElement;
+        }
+        while (element != null) {
+            if (element.hash == hash(key) && Objects.equals(key, element.key)) {
```

Comment:
if objects are equal, their hashcodes are equal too

Why do you compare hashs, if you already compare keys?